### ISSUE TITLE
Option parameters not being sent to subscriber

### ISSUE THREAD

**Issue by Jeferson-Menegazzo** [typeorm/typeorm#2431]
**Issue type:**

[ ] question
[X] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[X] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`

**TypeORM version:**

[X] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

Some methods of the ```Repository<T>``` like:

``` 
save<T extends DeepPartial<Entity>>(entity: T, options?: SaveOptions): Promise<T>;

remove(entity: Entity, options?: RemoveOptions): Promise<Entity>;

delete(criteria: string | string[] | number | number[] | Date | Date[] | ObjectID | ObjectID[] | FindConditions<Entity>, options?: RemoveOptions): Promise<DeleteResult>;
```

have ```Options``` object, where there is the ```data``` attribute, that can be used in subscribers. To access the Options in subscriber, I use ```event.queryRunner.data```. However, the ```delete``` method always passes undefined.

**Comment by pleerock** [comment]
what about `save`? Does it work properly?

**Comment by Jeferson-Menegazzo** [comment]
```save``` and ```remove``` work properly, just ```delete``` method don't, passing ```RemoveOptions``` ```undefined```.

**Comment by pleerock** [comment]
Feel free to contribute! Just take a look how save passes this data down and figure out why remove doesn't do the same

**Comment by rustamwin** [comment]
> save and remove work properly, just delete method don't.

because `delete` method executes a primitive operation without functionalities

**Comment by rustamwin** [comment]
Fixed #3888 
In the next release will be removed
